src/main/java/net/imglib2/img/display/imagej/ImageJVirtualStack.java

```diff
@@ -34,65 +34,68 @@
 
 package net.imglib2.img.display.imagej;
 
-import java.util.concurrent.ExecutorService;
-import java.util.stream.IntStream;
-import java.util.stream.LongStream;
-
-import net.imglib2.Cursor;
 import net.imglib2.Interval;
+import net.imglib2.Positionable;
 import net.imglib2.RandomAccessibleInterval;
 import net.imglib2.converter.Converter;
+import net.imglib2.converter.Converters;
 import net.imglib2.display.projector.AbstractProjector2D;
 import net.imglib2.display.projector.IterableIntervalProjector2D;
 import net.imglib2.display.projector.MultithreadedIterableIntervalProjector2D;
 import net.imglib2.img.Img;
 import net.imglib2.img.array.ArrayImg;
 import net.imglib2.img.array.ArrayImgFactory;
-import net.imglib2.img.array.ArrayImgs;
 import net.imglib2.img.basictypeaccess.array.ArrayDataAccess;
 import net.imglib2.type.NativeType;
-import net.imglib2.type.numeric.RealType;
 import net.imglib2.util.IntervalIndexer;
+import net.imglib2.util.Util;
 import net.imglib2.view.Views;
 
+import java.util.concurrent.ExecutorService;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+
 /**
  * TODO
  *
  */
 public class ImageJVirtualStack< S, T extends NativeType< T > > extends AbstractVirtualStack
 {
+
 	final private long[] higherSourceDimensions;
 
-	final private RandomAccessibleInterval< S > source;
+	final private RandomAccessibleInterval< T > source;
 
 	private final T type;
 
-	private final Converter< ? super S, T > converter;
-
 	private boolean isWritable = false;
 
-	final protected ExecutorService service;
+	protected ExecutorService service;
 
 	/* old constructor -> non-multithreaded projector */
 	protected ImageJVirtualStack( final RandomAccessibleInterval< S > source, final Converter< ? super S, T > converter,
 			final T type, final int bitDepth )
 	{
-		this( source, converter, type, bitDepth, null );
+		this( Converters.convert( source, converter, type ), bitDepth );
 	}
 
 	protected ImageJVirtualStack( final RandomAccessibleInterval< S > source, final Converter< ? super S, T > converter,
 			final T type, final int bitDepth, final ExecutorService service )
+	{
+		this( source, converter, type, bitDepth );
+		setExecuterService(service);
+	}
+
+	protected ImageJVirtualStack( final RandomAccessibleInterval< T > source, final int bitDepth )
 	{
 		super( ( int ) source.dimension( 0 ), ( int ) source.dimension( 1 ), multiply( initHigherDimensions( source ) ), bitDepth );
 		// if the source interval is not zero-min, we wrap it into a view that
 		// translates it to the origin
 		// if we were given an ExecutorService, use a multithreaded projector
 		assert source.numDimensions() > 1;
 		this.source = zeroMin( source );
+		this.type = Util.getTypeFromInterval( source );
 		this.higherSourceDimensions = initHigherDimensions( source );
-		this.type = type;
-		this.converter = converter;
-		this.service = service;
 	}
 
 	private static int multiply( final long[] higherSourceDimensions )
```

Comment:
This method can be removed. Use `Intervals.numElements( long... )` instead

Reply:
No it cannot be remove by `Intervals.numElements( long... )` because for an empty array numElements crashes, while `multiply` returns `1`. We could fix `Intervals.numElements(long ...)` to return `1` for an empty array?